app/controllers/internal/AccountTypeController.scala

```diff
@@ -63,12 +68,25 @@ trait AccountTypeController extends FrontendController with Actions {
 
   def createAuditContext(): TAuditContext
 
-  def accountTypeForCredId(credId: String) = Action.async { implicit request =>
-    val ruleContext = RuleContext(Some(credId))
-    val auditContext = createAuditContext()
-    ruleEngine.matchRulesForLocation(ruleContext, auditContext) map { location =>
-      val accountType = accountTypeBasedOnLocation(location)
-      Ok(Json.toJson(AccountTypeResponse(accountType)))
+  def createRuleContext(credId: String)(implicit hc: HeaderCarrier): RuleContext
+
+  def accountTypeForCredId(credId: String): Action[AnyContent] = Action.async { implicit request =>
+    val ruleContext = createRuleContext(credId)
+
+    ruleContext.affinityGroup.flatMap {
+      case AffinityGroupValue.AGENT =>
+        Future.successful(Ok(Json.toJson(AccountTypeResponse(AccountType.Agent))))
+      case _ =>
+        val auditContext = createAuditContext()
+        ruleEngine.matchRulesForLocation(ruleContext, auditContext) map { location =>
+          val accountType: AccountType = accountTypeBasedOnLocation(location)
+          Ok(Json.toJson(AccountTypeResponse(accountType)))
+        }
+    }.recover {
+      case e =>
+        e.printStackTrace()
+        logger.error(s"Failed to get user details due to downstream failure.", e)
+        InternalServerError("Failed to get user details due to downstream failure.")
```

Comment:
Not sure about the word "failure", but also not overly concerned. @dclifton-hmrc thoughts?